rst_files/linear_algebra.rst

```diff
@@ -183,41 +181,26 @@ Scalar multiplication is illustrated in the next figure
 
     # illustrate scalar multiplication
 
-    x = [2, 2]
-    scalars = [-2, 2]
-
-    # Create matrices of x and y values, labels for plotting
-    x_vals = zeros(2, 1 + length(scalars))
-    y_vals = zeros(2, 1 + length(scalars))
-    labels = []
-    x_vals[2, 3] = x[1]
-    y_vals[2, 3] = x[2]
-    labels = [labels; (x[1] + 0.4, x[2] - 0.2, "x")]
-
-    # Perform scalar multiplication, store results in plotting matrices
-    for i ∈ eachindex(scalars)
-        s = scalars[i]
-        v = s .* x
-        x_vals[2, i] = v[1]
-        y_vals[2, i] = v[2]
-        labels = [labels; (v[1] + 0.4, v[2] - 0.2, string(s, "x"))]
-    end
+    x = [2, 2, 2]
+    scalars = [-2, 1, 2]
+    vals = x .* scalars
+    x_vals = reduce(hcat, ([0, v] for v in vals))
+    y_vals = x_vals
+    labels = map((v, s) -> (v + 0.4, v - 0.2, string(isone(s) ? "" : s, "x")), vals, scalars)
```

Comment:
This is pretty tricky code for what we really need.  We can certainly use the `map`, as we taught it, but I don't think this is really worth it for this example.  Why not hardcode the 3 labels?

Reply:
We could hardcode the labels. Is a matter of whether is worthwhile for them to see how to generate the labels programatically.

Reply:
This is too complicated for intro users

Reply:
I will just hard-code all the values at least for the ones in this section.